### PR TITLE
Fix null args, variable shadowing.

### PR DESCRIPTION
**Pull request by brndnmtthws**
cc @elingg

### CODE REVIEW

File: src/main/scala/org/apache/mesos/chronos/scheduler/jobs/TaskUtils.scala [hunk 1]
```diff
@@ -90,8 +90,8 @@ object TaskUtils {
     })
   }
 
-  def getTaskId(job: BaseJob, due: DateTime, attempt: Int = 0, arguments: String = ""): String = {
-    taskIdTemplate.format(due.getMillis, attempt, job.name, arguments)
+  def getTaskId(job: BaseJob, due: DateTime, attempt: Int = 0): String = {
+    taskIdTemplate.format(due.getMillis, attempt, job.name, job.arguments.foldRight("")((a, b) => if (b.nonEmpty) s"$a $b" else a))
   }
```
Comment (elingg) [inline src/main/scala/org/apache/mesos/chronos/scheduler/jobs/TaskUtils.scala]: This makes sense, but could we simplify the logic some by iterating through the list from left to right and doing a simple string concatenation. job.arguments.foldLeft("") ((a, b) => if (b.nonEmpty) a + " " + b else a))

**Comment by elingg** [comment]
Thank you for fixing this. This should fix https://github.com/mesos/chronos/issues/355. I have only one comment above.

File: src/main/scala/org/apache/mesos/chronos/scheduler/jobs/TaskUtils.scala (hunk 1 above)
Comment (brndnmtthws) [inline src/main/scala/org/apache/mesos/chronos/scheduler/jobs/TaskUtils.scala]: You're probably right. I got a bit carried away.

**Comment by tamizhgeek** [comment]
Thanks guys!

**Pull request merged**